src/util/time.rs

```diff
@@ -281,17 +281,19 @@ impl Instant {
     // Use millisecond resolution for ignoring the error.
     fn elapsed_duration_coarse(later: Timespec, earlier: Timespec) -> Duration {
         let later_ms =
-            later.sec * MILLISECOND_PER_SECOND + (later.nsec / NANOSECONDS_PER_MILLISECOND) as i64;
+            later.sec * MILLISECOND_PER_SECOND + later.nsec as i64 / NANOSECONDS_PER_MILLISECOND;
         let earlier_ms = earlier.sec * MILLISECOND_PER_SECOND +
-            (earlier.nsec / NANOSECONDS_PER_MILLISECOND) as i64;
-        if later_ms >= earlier_ms {
-            Duration::from_millis((later_ms - earlier_ms) as u64)
+            earlier.nsec as i64 / NANOSECONDS_PER_MILLISECOND;
+        let dur = later_ms - earlier_ms;
+        if dur >= 0 {
+            Duration::from_millis(dur as u64)
         } else {
-            panic!(
-                "system time jumped back, {:.3} -> {:.3}",
+            warn!(
```

Comment:
If the jump back is really due to cpu switching, then I can foresee that we are going to remove this warning in the future.